craft_parts/executor/executor.py

```diff
@@ -91,7 +91,7 @@ def prologue(self) -> None:
 
         # update the overlay environment package list to allow installation of
         # overlay packages.
-        if any(p.has_overlay for p in self._part_list):
+        if any(p.spec.overlay_packages for p in self._part_list):
```

Comment:
not sure if this is an 'overreach' from the executor; it might be preferrable to add a `has_overlay_packages` method/property in the Part (like `has_overlay`)

Reply:
That should be fine for now.